src/components/Editor/Editor.tsx

```diff
@@ -1,116 +1,184 @@
-    setIsEditorLoading(false);
-    effectCalled.current = true;
-  }, [language, defaultValue, workerPaths, options]);
-
-  useEffect(() => {
-    isEditorLoading && createEditor();
-  }, [isEditorLoading, createEditor]);
-
-  useEffect(() => {
-    modelChangeEffect.current?.dispose();
-    modelChangeEffect.current = editorRef.current?.onDidChangeModel(() => editorRef.current.focus());
-  }, []);
-
-  return (
-    <div className="editor-wrapper" style={{ display: "flex", flexFlow: "column nowrap", height: "100%" }}>
-      {!!editorRef.current && <EditorLauncher {...launcherProps} />}
-      <div
-        hidden={isEditorLoading}
-        ref={containerRef}
-        className="editor-container"
-        style={{
-          maxHeight: 916,
-          minWidth: "100%",
-          width: width ?? "100%",
-          height: height ?? 256,
-          borderTop: "1px solid",
-          borderBottom: "1px solid",
-          borderColor: "#52525226",
-          flex: 2,
-          ...style,
-        }}
-        {...props}></div>
-      {isEditorLoading && <>isLoading...</>}
-    </div>
-  );
-};
+    useEffectOnce(() => {
+      if (editorRef.current) {
+        store.saveTabSession(editorRef.current, defaultSessionId);
+        editorRef.current.dispose();
+        setIsEditorLoading(true);
+      }
+    });
+
+    const createEditor = useCallback(() => {
+      if (!containerRef.current || effectCalled.current) return;
+      setWorkers(workerPaths);
+
+      const model = defaultSessionId
+        ? store.getSessionState(defaultSessionId)?.model ?? store.createTabSession(defaultSessionId, { language, value: defaultValue }).model
+        : EditorStoreManager.createModel({ language, value: defaultValue });
+
+      const state = defaultSessionId && store.getSessionState(defaultSessionId).state;
+
+      // create monaco-editor instance
+      editorRef.current = monaco.editor.create(containerRef.current, {
+        model,
+        minimap: { enabled: false },
+        autoIndent: "full",
+        theme: "thanosql-light",
+        mouseWheelZoom: true,
+        fontSize: 16,
+        inDiffEditor: false,
+        renderLineHighlight: "none",
+        lineNumbers: ln => '<span style="padding-left: 16px; color: #C7C9CC;">' + ln + "</span>",
+        glyphMargin: false,
+        folding: false,
+        lineDecorationsWidth: "32px",
+        lineNumbersMinChars: 0,
+        detectIndentation: true,
+        tabSize: 4,
+        automaticLayout: true,
+        ...options,
+      });
+
+      if (state) editorRef.current.restoreViewState(state);
+      setIsEditorLoading(false);
+      effectCalled.current = true;
+    }, [language, defaultValue, workerPaths, options]);
+
+    const setEditorRef = useCallback(() => {
+      editorRefs.current[editorId] = modules;
+      const newActiveEditors = Object.keys(editorRefs.current);
+      setActiveEditors(prev => new Set([...prev, ...newActiveEditors]));
```

Comment:
```suggestion
      setActiveEditors(prev => new Set([...prev, editorId]));
```
or

131 line과 마찬가지로
```suggestion
      setActiveEditors(new Set(Object.keys(editorRefs.current)));
```

이렇게 해도 의도 대로 돌아가는지 궁금합니다